arch/x86_64/mcount-dynamic.c

```diff
@@ -141,16 +146,16 @@ static void read_xray_map(struct arch_dynamic_info *adi,
 	elf_get_secdata(elf, iter);
 	elf_read_secdata(elf, iter, 0, adi->xrmap, shdr->sh_size);
 
-	/* handle position independent code */
-	if (elf->ehdr.e_type == ET_DYN) {
-		struct xray_instr_map *xrmap;
-		unsigned i;
-
-		for (i = 0; i < adi->xrmap_count; i++) {
-			xrmap = &adi->xrmap[i];
+	for (i = 0; i < adi->xrmap_count; i++) {
+		xrmap = &adi->xrmap[i];
 
-			xrmap->addr  += offset;
-			xrmap->entry += offset;
+		if (xrmap->version == 2) {
+			xrmap->address += offset + (shdr->sh_offset + i * sizeof(struct xray_instr_map));
+			xrmap->function += offset + (shdr->sh_offset + i * sizeof(struct xray_instr_map) + 8);
```

Comment:
I prefer passing the variable to the sizeof like `sizeof(*xrmap)` as it's shorter and easier to deal with renames.